### PR TITLE
Close #3329 Update prepare repository for minor release action so that it updates the main branch version alias as a pull request.

### LINKED ISSUE #3329: Update prepare repository for minor release action so that it updates the main branch version alias as a pull request.
During the last two minor releases we needed to manually create a PR against main to update the dev branch alias.
See
- #3328
- #3053

See https://github.com/az-digital/az_quickstart/blob/main/.github/workflows/prepare-for-minor-release.yml

### PR DESCRIPTION
**Pull request by trackleft**
## Description
Update array syntax in workflow file
Add main to the matrix that creates pull requests for updating the dev branch alias.
Add           `- 2.10.x` and `main` branches to auto PR creation matrix.

## Related issues
Close #3329

## How to test

Build up a quickstart ecosystem under your own github namespace, and create a minor release (or just wait until 2.11.x)

### Arizona Quickstart (install profile, custom modules, custom theme)
- **Patch release changes**
   - [ ] Bug fix
   - [ ] Accessibility, performance, or security improvement
   - [ ] Critical institutional link or brand change
   - [ ] Adding experimental module
   - [ ] Update experimental module
- **Minor release changes**
   - [ ] New feature
   - [ ] Breaking or visual change to existing behavior
   - [ ] Upgrade experimental module to stable
   - [ ] Enable existing module by default or database update
   - [ ] Non-critical brand change
   - [ ] New internal API or API improvement with backwards compatibility
   - [ ] Risky or disruptive cleanup to comply with coding standards
   - [ ] High-risk or disruptive change (requires upgrade path, risks regression, etc.)
- **Other or unknown**
   - [ ] Other or unknown

### Drupal core
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch level release (non-security bug-fix release)
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major or minor level update
- **Other or unknown**
   - [ ] Other or unknown

### Drupal contrib projects
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch or minor level update
   - [ ] Add new module
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major level update
- **Other or unknown**
   - [ ] Other or unknown

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: .github/workflows/prepare-for-minor-release.yml [hunk 1]
```diff
@@ -137,14 +137,26 @@ jobs:
           .lando.yml
           .probo.yaml
           composer.json
-          
+
   update_previous_release_branches:
     needs: [prepare_for_new_minor_release, branch_name_prep]
     runs-on: ubuntu-latest
     name: Update `${{ matrix.previous_version }}` release branch.
```
Comment (trackleft) [inline .github/workflows/prepare-for-minor-release.yml:144]: `name: Update `${{ matrix.previous_version }}` release branch.` is probably not what we want to see in the PR for main.

File: .github/workflows/prepare-for-minor-release.yml [hunk 2]
```diff
@@ -174,3 +185,34 @@ jobs:
...
+        run: |
+          composer config extra.branch-alias.dev-main ${{ needs.branch_name_prep.outputs.next_minor_release }}.x-dev
+
+      - name: Create pull request for main branch updates
+        uses: peter-evans/create-pull-request@v6
+        with:
+          title: Prepare `main` branch for new minor release branch ${{ inputs.release_branch_name }}
+          commit-message: Prepare `main` branch for new minor release branch ${{ inputs.release_branch_name }}
+          branch: update-main-for-${{ inputs.release_branch_name }}
+          base: main
```
Comment (trackleft) [inline .github/workflows/prepare-for-minor-release.yml:214]: Not sure what label to put here, since we don't have access to the next version number yet.

Comment (trackleft) [reply .github/workflows/prepare-for-minor-release.yml:214]: The other step sets a label like this
```
          labels: |
            ${{ matrix.previous_version }} only	            ${{ matrix.previous_version }} only
```

Comment (joeparsons) [reply .github/workflows/prepare-for-minor-release.yml:214]: Maybe just "task"?

Comment (trackleft) [inline .github/workflows/prepare-for-minor-release.yml:214]:
```suggestion
          base: main
          labels: |
            task
```

**Review by joeparsons — APPROVED** [review]

**Review by tadean — APPROVED** [review]

**Pull request merged**